### ISSUE TITLE
Getting errors while running retention time correction

### ISSUE THREAD

**Issue by raagbtitl** [sneumann/xcms#345]
I am working with mzXML files generated frpm the the ACQUITY UPLC  (XEVO-G2SQTOF) .

**Getting following error**s:

> s <- adjustRtime(res, param = ObiwarpParam())
Sample number 3 used as center sample.

Aligning Mouse_feces C18_NEG_sa1.mzXML against Mouse_feces_C18_NEG_sa1_inj2.mzXML ...
Error in names(res) <- nms :
  'names' attribute [5] must be the same length as the vector [4]
In addition: Warning message:
stop worker failed:
  'clear_cluster' receive data failed:
  reached elapsed time limit
>
> pgp <- PeakGroupsParam(minFraction = 1, extraPeaks = 1, smooth = "loess",
+   span = 0.2, family = "gaussian")
> s <- adjustRtime(res, param = pgp)
Error in `colnames<-`(`*tmp*`, value = basename(fileNames(object))) :
  attempt to set 'colnames' on an object with less than two dimensions
> pgp <- PeakGroupsParam(minFraction = 0.5, extraPeaks = 1, smooth = "loess", span = 0.2, family = "gaussian")
> s <- adjustRtime(res, param = pgp)
Performing retention time correction using 28 peak groups.
Error in do_adjustRtime_peakGroups(chromPeaks(object), peakIndex = featureDefinitions(object)$peakidx,  :
  Not enough peak groups even for linear smoothing available!
In addition: Warning message:
In do_adjustRtime_peakGroups(chromPeaks(object), peakIndex = featureDefinitions(object)$peakidx,  :
  Too few peak groups for 'loess', reverting to linear method

> res
MSn experiment data ("XCMSnExp")
Object size in memory: 8.2 Mb
- - - Spectra data - - -
 MS level(s): 1
 Number of spectra: 34080
 MSn retention times: 0:1 - 29:60 minutes
- - - Processing information - - -
Data loaded [Wed Dec  5 12:51:52 2018]
Filter: select MS level(s) 1 [Wed Dec  5 12:51:52 2018]
 MSnbase version: 2.7.3
- - - Meta data  - - -
phenoData
  rowNames: 1 2 ... 6 (6 total)
  varLabels: sample_name sample_group
  varMetadata: labelDescription
Loaded from:
  [1] blank_C18_NEG_23Oct18_3.mzXML...  [6] Mouse_feces_C18_NEG_sa2_inj2.mzXML
  Use 'fileNames(.)' to see all files.
protocolData: none
featureData
  featureNames: F1.S00008 F1.S00011 ... F6.S14680 (34080 total)
  fvarLabels: fileIdx spIdx ... spectrum (29 total)
  fvarMetadata: labelDescription
experimentData: use 'experimentData(object)'
- - - xcms preprocessing - - -
Chromatographic peak detection:
 method: centWave
 224 peaks identified in 6 samples.
 On average 44.8 chromatographic peaks per sample.
Correspondence:
 method: chromatographic peak density
 56 features identified.
 Median mz range of features: 0
 Median rt range of features: 3.7725

> sessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
 [1] tools     stats4    parallel  stats     graphics  grDevices utils
 [8] datasets  methods   base

other attached packages:
 [1] dplyr_0.7.6         optparse_1.6.0      magrittr_1.5
 [4] RColorBrewer_1.1-2  MAIT_1.15.0         pls_2.6-0
 [7] CAMERA_1.37.0       xcms_3.4.1          MSnbase_2.8.2
[10] ProtGenerics_1.13.0 S4Vectors_0.19.19   mzR_2.15.1
[13] Rcpp_0.12.18        Biobase_2.41.2      BiocGenerics_0.27.1
[16] BiocParallel_1.15.8

loaded via a namespace (and not attached):
  [1] backports_1.1.2        spam_2.2-0             Hmisc_4.1-1
  [4] plyr_1.8.4             igraph_1.2.2           lazyeval_0.2.1
  [7] sp_1.3-1               splines_3.5.0          AlgDesign_1.1-7.3
 [10] ggplot2_3.0.0          digest_0.6.15          foreach_1.4.4
 [13] BiocInstaller_1.31.3   htmltools_0.3.6        gdata_2.18.0
 [16] checkmate_1.8.5        cluster_2.0.7-1        doParallel_1.0.11
 [19] sfsmisc_1.1-2          limma_3.37.3           recipes_0.1.3
 [22] gower_0.1.2            gmodels_2.18.1         dimRed_0.1.0
 [25] colorspace_1.3-2       crayon_1.3.4           graph_1.59.0
 [28] bindr_0.1.1            impute_1.55.0          survival_2.42-6
 [31] iterators_1.0.10       glue_1.3.0             DRR_0.0.3
 [34] gtable_0.2.0           ipred_0.9-6            zlibbioc_1.27.0
 [37] questionr_0.6.3        kernlab_0.9-27         ddalpha_1.3.4
 [40] DEoptimR_1.0-8         maps_3.3.0             abind_1.4-5
 [43] scales_1.0.0           vsn_3.49.1             miniUI_0.1.1.1
 [46] xtable_1.8-2           spData_0.2.9.3         htmlTable_1.12
 [49] magic_1.5-8            foreign_0.8-71         spdep_0.7-7
 [52] preprocessCore_1.43.0  dotCall64_1.0-0        Formula_1.2-3
 [55] lava_1.6.3             prodlim_2018.04.18     getopt_1.20.2
 [58] htmlwidgets_1.2        gplots_3.0.1           acepack_1.4.1
 [61] pkgconfig_2.0.1        XML_3.98-1.16          deldir_0.1-15
 [64] nnet_7.3-12            caret_6.0-80           tidyselect_0.2.4
 [67] rlang_0.2.1            reshape2_1.4.3         later_0.7.3
 [70] munsell_0.5.0          broom_0.5.0            geometry_0.3-6
 [73] stringr_1.3.1          mzID_1.19.0            ModelMetrics_1.2.0
 [76] RhpcBLASctl_0.18-205   knitr_1.20             robustbase_0.93-2
 [79] caTools_1.17.1.1       purrr_0.2.5            RANN_2.6
 [82] bindrcpp_0.2.2         RBGL_1.57.0            nlme_3.1-137
 [85] mime_0.5               RcppRoll_0.3.0         compiler_3.5.0
 [88] rstudioapi_0.7         e1071_1.7-0            affyio_1.51.0
 [91] klaR_0.6-14            MassSpecWavelet_1.47.0 tibble_1.4.2
 [94] stringi_1.2.4          highr_0.7              fields_9.6
 [97] lattice_0.20-35        Matrix_1.2-14          multtest_2.37.0
[100] LearnBayes_2.15.1      pillar_1.3.0           combinat_0.0-8
[103] MALDIquant_1.18        data.table_1.11.4      bitops_1.0-6
[106] httpuv_1.4.5           agricolae_1.2-8        R6_2.2.2
[109] latticeExtra_0.6-28    pcaMethods_1.73.0      affy_1.59.0
[112] promises_1.0.1         KernSmooth_2.23-15     gridExtra_2.3
[115] IRanges_2.15.16        codetools_0.2-15       boot_1.3-20
[118] MASS_7.3-50            gtools_3.8.1           assertthat_0.2.0
[121] CVST_0.2-2             withr_2.1.2            expm_0.999-2
[124] plsgenomics_1.5-1      grid_3.5.0             rpart_4.1-13
[127] timeDate_3043.102      coda_0.19-1            tidyr_0.8.1
[130] class_7.3-14           shiny_1.1.0            lubridate_1.7.4
[133] base64enc_0.1-3

**Comment by jotsetung** [comment]
Yes, I guess for peak group alignment you have simply to few peaks in your data. Regarding obiwarp, I have no immediate solution/explanation what might have happened. You could try with a different `binSize` (e.g. `ObiwarpParam(binSize = 0.4)`.

To really track the problem down I would need to get hand on the data.
